### PR TITLE
Add geopoint object to model

### PR DESCRIPTION
**Pull request by gbouv**

### CODE REVIEW

File: lafourche/model/geopoint.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    def __init__(self, longitude: float, latitude: float):
+        self.lat = latitude
+        self.lon = longitude
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Geopoint):
+            return False
+        return o.lon == self.lon and o.lat == self.lat
+
+    def __str__(self) -> str:
```
Comment (margueritevial) [inline lafourche/model/geopoint.py]: we could only use __repr__ instead of __str__, because it is unambiguous.

File: lafourche/model/geopoint.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        return "Geopoint(lon: " + str(self.lon) + ", lat: " + str(self.lat) + ")"
+
+    def __repr__(self) -> str:
```
Comment (margueritevial) [inline lafourche/model/geopoint.py:16]: can we make sure to add the __repr__ method to all other classes (for example Edge)

File: lafourche/model/node.py [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
-        self.longitude = longitude
+        self.coord = coord
 
     def __eq__(self, o: object) -> bool:
         if not isinstance(o, Node):
             return False
-        return o.latitude == self.latitude and o.longitude == self.longitude and o.__node_id == self.__node_id
+        return o.coord == self.coord and o.__node_id == self.__node_id
 
     def __str__(self) -> str:
```
Comment (margueritevial) [inline lafourche/model/node.py]: we can remove __str__ and use __repr__ instead

File: lafourche/parser/osmparser.py [hunk 4]
```diff
@@ -65,8 +66,8 @@ def get_node_registry(self, tree_root: ElementTree) -> {int: Node}:
                                   node.tag, node.attrib.get('id'), node.attrib.get('lat'), node.attrib.get('lon'))
                 continue
             else:
-                node_registry[node.attrib.get('id')] = Node(node.attrib.get('id'), node.attrib.get('lon'),
-                                                            node.attrib.get('lat'))
+                node_registry[node.attrib.get('id')] = Node(node.attrib.get('id'),
+                                                            Geopoint(node.attrib.get('lon'), node.attrib.get('lat')))
```
Comment (margueritevial) [inline lafourche/parser/osmparser.py:70]: this is a bug because Geopoint parameters should be float instead of str here

**Review by margueritevial — APPROVED** [review]
some minor comments but LGTM

File: lafourche/parser/osmparser.py (hunk 4 above)
Comment (gbouv) [reply lafourche/parser/osmparser.py:70]: Fixed in #13

**Review by margueritevial — APPROVED** [review]

**Review by encoreunefeuille — APPROVED** [review]
lgtm from encoreunefeuille ;)

**Pull request merged**